### PR TITLE
[Playwright] PEPPER-842 DSM Participant List filter test

### PR DESCRIPTION
**Pull request by aweng98**
[PEPPER-842](https://broadworkbench.atlassian.net/browse/PEPPER-842)

New tests:

- `playwright-e2e/tests/dsm/participant-list/filter-custom-view.spec.ts`
- `playwright-e2e/tests/dsm/participant-list/filter-short-id.spec.ts`
- `playwright-e2e/tests/dsm/participant-list/filter-registration-date.spec.ts`
- 

[PEPPER-842]: https://broadworkbench.atlassian.net/browse/PEPPER-842?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: playwright-e2e/tests/dsm/kitUploadFlow/blood-kit-upload-flow.spec.ts [hunk 1]
```diff
@@ -49,7 +51,9 @@ test.describe.parallel('Blood Kits upload flow', () => {
   });
 
   for (const study of studies) {
-    test(`Should upload a single kit for one participant @functional @visual @dsm @${study}`, async () => {
+    test(`Should upload a single kit for one participant @functional @visual @dsm @${study}`, async ({page}, testInfo) => {
+      testResultDir = testInfo.outputDir;
```
Comment (aweng98) [inline playwright-e2e/tests/dsm/kitUploadFlow/blood-kit-upload-flow.spec.ts:55]: [testInfo.outputDir](https://www.cuketest.com/playwright/docs/api/class-testinfo#test-info-output-dir)

File: playwright-e2e/utils/log-utils.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { test } from '@playwright/test';
 
-export function logParticpantCreated(participantEmail: string, participantName: string) {
+export function logParticipantCreated(participantEmail: string, participantName: string) {
```
Comment (aweng98) [inline playwright-e2e/utils/log-utils.ts:3]: Typo

File: playwright-e2e/utils/test-utils.ts [hunk 3]
```diff
@@ -18,7 +18,10 @@ export async function waitForNoSpinner(page: Page): Promise<void> {
 
 export async function waitForResponse(page: Page, {uri, status = 200, timeout = 30000}: WaitForResponse): Promise<Response> {
   try {
-    return await page.waitForResponse((response: Response) => response.url().includes(uri) && response.status() === status, {timeout})
+    return page.waitForResponse(
+      (response: Response) => response.url().includes(uri) && response.status() === status,
+      {timeout}
+    )
```
Comment (aweng98) [inline playwright-e2e/utils/test-utils.ts:24]: formating

File: playwright-e2e/fixtures/dsm-fixture.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+// Use this fixture for login in DSM tests
+const fixture = base.extend<Fixtures>({
+  page: async ({ page }, use) => {
+    await login(page);
+    await use(page);
+  }
+});
+
+export const test = fixture;
```
Comment (aweng98) [inline playwright-e2e/fixtures/dsm-fixture.ts:13]: Reduces duplicates of common code.

File: playwright-e2e/lib/component/dsm/filters/sections/customize-view.ts [hunk 5]
```diff
@@ -67,10 +77,11 @@ export class CustomizeView {
 
   /* XPaths */
   private get openButtonXPath(): string {
-    return (
-      "//div[text()[normalize-space()='Customize View'] and " +
-      "button[.//*[local-name()='svg' and @data-icon='server']/*[local-name()='path']]]/button"
-    );
+    return `//*[text()[normalize-space()="Customize View"]]/button`;
```
Comment (aweng98) [inline playwright-e2e/lib/component/dsm/filters/sections/customize-view.ts:80]: Simplified xpath

File: playwright-e2e/lib/component/dsm/filters/sections/customize-view.ts [hunk 6]
```diff
@@ -67,10 +77,11 @@ export class CustomizeView {
...
+  }
+
+  private async isPanelOpen(): Promise<boolean> {
+    return await this.page.locator('.btn-group').count() >= 1;
```
Comment (aweng98) [inline playwright-e2e/lib/component/dsm/filters/sections/customize-view.ts:84]: This function helps to determine whether Customize View panel is open or closed.

File: playwright-e2e/lib/component/dsm/filters/sections/search/search.ts [hunk 7]
```diff
@@ -46,10 +46,15 @@ export class Search {
...
+    // Click plus icon to expand Date fields
+    //const expandDateRangeButton = this.page.locator(`xpath=${this.baseColumnXPath(columnName)}//button[.//*[@data-icon="plus-square"]]`);
+    //await expandDateRangeButton.click();
+
+    const dateInputFields = this.page.locator(this.dateInputFieldXPath(columnName));
+    await expect(dateInputFields).toHaveCount(2); // synchronize with UI
 
     fromDate && (await dateInputFields.nth(0).fill(fromDate));
-    toDate && (await dateInputFields.nth(1).fill(fromDate));
+    toDate && (await dateInputFields.nth(1).fill(toDate));
```
Comment (aweng98) [inline playwright-e2e/lib/component/dsm/filters/sections/search/search.ts:57]: Bug fix

File: playwright-e2e/lib/component/dsm/tables/participantListTable.ts [hunk 8]
```diff
@@ -1,10 +1,13 @@
 import { Locator, Page } from '@playwright/test';
+import Table from 'lib/component/table';
 import ParticipantPage from 'pages/dsm/participant-page/participant-page';
 
-export class ParticipantListTable {
+export class ParticipantListTable extends Table {
```
Comment (aweng98) [inline playwright-e2e/lib/component/dsm/tables/participantListTable.ts:5]: Refactored `Table` to support DSM tables such as Participant List table.

File: playwright-e2e/pages/dsm/kitUpload-page/kitUpload-page.ts [hunk 9]
```diff
@@ -26,8 +26,9 @@ export default class KitUploadPage {
     await this.kitType.selectKitType(kitType);
   }
 
-  public async uploadFile(kitType: KitTypeEnum, kitInfo: KitUploadInfo[], study: StudyEnum) {
-    const path = `${__dirname}/${kitType}_${study}-${new Date().getTime()}.txt`;
+  public async uploadFile(kitType: KitTypeEnum, kitInfo: KitUploadInfo[], study: StudyEnum, testResultDir?: string) {
+    const dir = testResultDir ? testResultDir : __dirname;
```
Comment (aweng98) [inline playwright-e2e/pages/dsm/kitUpload-page/kitUpload-page.ts:30]: Use `testInfo.outputDir` so new text files are inside a git-ignored dir.

File: playwright-e2e/pages/dsm/participant-list-page.ts [hunk 10]
```diff
@@ -0,0 +1,134 @@
+import { APIRequestContext, expect, Locator, Page } from '@playwright/test';
+import Modal from 'lib/component/dsm/modal';
+import { StudyNavEnum } from 'lib/component/dsm/navigation/enums/studyNav-enum';
+import { Navigation } from 'lib/component/dsm/navigation/navigation';
+import { WelcomePage } from 'pages/dsm/welcome-page';
+import { waitForNoSpinner, waitForResponse } from 'utils/test-utils';
+import { Filters } from 'lib/component/dsm/filters/filters';
+import { ParticipantListTable } from 'lib/component/dsm/tables/participantListTable';
+
+export default class ParticipantListPage {
```
Comment (aweng98) [inline playwright-e2e/pages/dsm/participant-list-page.ts:10]: Renamed file from `participantList-page.ts` to `participant-list-page.ts`.

File: playwright-e2e/pages/page-base.ts [hunk 11]
```diff
@@ -328,10 +328,11 @@ export default abstract class PageBase implements PageInterface {
...
   async fillInFullName(fullName: string, opts?: { testId: string }): Promise<void> {
-    if (!opts) {
-      return this.page.getByRole('combobox', { name: 'Full Name' }).fill(fullName);
-    }
-    await this.page.getByTestId(opts.testId).fill(fullName);
+    const waitForResponsePromise = this.page.waitForResponse(response => response.status() === 200);
+    await Promise.all([
+      opts ? this.page.getByTestId(opts.testId).fill(fullName) : this.page.getByRole('combobox', { name: 'Full Name' }).fill(fullName),
+      waitForResponsePromise
+    ]);
```
Comment (aweng98) [inline playwright-e2e/pages/page-base.ts:335]: Add `waitForResponse` check when typing.

File: playwright-e2e/lib/component/dsm/filters/sections/search/search.ts (hunk 7 above)
Comment (GiCharkviani) [reply playwright-e2e/lib/component/dsm/filters/sections/search/search.ts:57]: Good catch! Thanks

File: playwright-e2e/lib/component/dsm/modal.ts [hunk 12]
```diff
@@ -0,0 +1,42 @@
+import { Locator, Page } from '@playwright/test';
```
Comment (GiCharkviani) [inline playwright-e2e/lib/component/dsm/modal.ts:1]: Very useful object. Thanks for creating that

**Review by GiCharkviani — APPROVED** [review]
LOOKS GREAT TO ME! 

I should refactor some of the DSM code after this PR merge, I guess :D 

Thanks for creating some useful features

**Review by aweng98 — COMMENTED** [review]
Fixed problem of CircleCI VMs running duplicated tests.

**Pull request merged**